src/csharp/Tokenizer.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.ML.OnnxRuntimeGenAI
+{
+    public class Tokenizer : IDisposable
+    {
+        private IntPtr _tokenizerHandle;
+        private bool _disposed = false;
+
+        public Tokenizer(Model model)
+        {
+            Result.VerifySuccess(NativeMethods.OgaCreateTokenizer(model.Handle, out _tokenizerHandle));
+        }
+
+        public Sequences EncodeBatch(string[] strings)
+        {
+            IntPtr stringArray = IntPtr.Zero;
+            UIntPtr[] stringLengths = new UIntPtr[strings.Length];
+            for (int i = 0; i < strings.Length; i++)
+            {
+                stringLengths[i] = (UIntPtr)UTF8Encoding.UTF8.GetByteCount(strings[i]);
+            }
+
+            unsafe
+            {
+                fixed (UIntPtr* stringLengthsPtr = stringLengths)
+                {
+                    Result.VerifySuccess(NativeMethods.OgaCreateAllocatedStrings((UIntPtr)strings.Length, stringLengthsPtr, out stringArray));
+                }
```

Comment:
Wondering why we need this array that would get destroyed at the end of the function? Can we do without it?
Can we pass strings one by one and encode it directly to the result?

Reply:
Changed the code so I now do the EncodeBatch and DecodeBatch by calling Encode and Decode in a loop. Was able to remove the OgaStrings type as a result of this.